analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedStringBuilder.cs

```diff
@@ -0,0 +1,77 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class UnusedStringBuilder : UnusedStringBuilderBase<SyntaxKind, VariableDeclaratorSyntax, IdentifierNameSyntax>
+{
+    private static readonly SyntaxKind[] SkipChildren =
+    {
+        SyntaxKind.ClassDeclaration,
+        SyntaxKind.StructDeclaration,
+        SyntaxKind.EnumDeclaration,
+        SyntaxKind.InterfaceDeclaration,
+        SyntaxKind.UsingDirective,
```

Comment:
Why is this?

Reply:
In the case of top-level statements we are not going through the children of the using directives (see [here](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgAQAYACfARgDoAVAUwA8EBuYgeheIB4BaL4gOQD2CYgHMBsEcVgooGGsQQALKCgCwAKA0A3CACdiKYMQC8xGDQDuxAMoJdEgEIBXKABs5ugBQBKRkA==)).

- The test for top-level statements is called _UnusedStringBuilder_TopLevelStatements_ and it's a parameterized test.